brack/src/main.rs

```diff
@@ -1,31 +1,8 @@
 use std::{fs::read_dir, path::Path};
 
 use anyhow::Result;
-use clap::{Parser, Subcommand};
-
-#[derive(Debug, Subcommand)]
-enum SubCommands {
-    #[clap(arg_required_else_help = true)]
-    Compile {
-        #[clap(short, long)]
-        plugins_dir_path: Option<String>,
-
-        #[clap(short, long)]
-        backend: String,
-
-        #[clap(short, long)]
-        filename: String,
-    },
-    Build,
-    LanguageServer,
-    New {
-        #[clap(short, long)]
-        name: String,
-    },
-    Add {
-        schema: String,
-    },
-}
+use brack::sub_commands::SubCommands;
+use clap::Parser;
 
 #[derive(Parser, Debug)]
 struct Args {
```

Comment:
Why is only SubCommands moved to another file and Arg left in main?

Reply:
To be honest, I moved `SubCommands` to check caches to work, there are no other reasons. However, even if I consider the design, we don't need to treat `Args` specially, because it's only needed by the clap parser generator, and we won't be defining its methods. If this split bothers you, I would revert this commit.

Reply:
Understood. I thought it was a mistake, but now that I know there's a valid reason, I'm convinced it's fine as it is.